app/Console/Commands/MaintenanceFetchOuis.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Console\LnmsCommand;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\DB;
+use LibreNMS\Config;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MaintenanceFetchOuis extends LnmsCommand
+{
+    /**
+     * The name of the console command.
+     *
+     * @var string
+     */
+    protected $name = 'maintenance:fetch-ouis';
+
+    protected string $mac_oui_url = 'https://www.wireshark.org/download/automated/data/manuf';
+    protected int $min_refresh_days = 6;
+    protected int $max_wait_seconds = 900;
+    protected int $upsert_chunk_size = 1000;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->addOption('force', null, InputOption::VALUE_NONE);
+        $this->addOption('wait', null, InputOption::VALUE_NONE);
+    }
+
+    /**
+     * Execute the console command.
+     */
+    public function handle(): int
+    {
+        $force = $this->option('force');
+
+        if (Config::get('mac_oui.enabled') !== true && ! $force) {
+            $this->line(trans('commands.maintenance:fetch-ouis.disabled', ['setting' => 'mac_oui.enabled']));
+
+            if (! $this->confirm(trans('commands.maintenance:fetch-ouis.enable_question'))) {
+                return 0;
+            }
+
+            Config::persist('mac_oui.enabled', true);
+        }
+
+        // We want to refresh after at least 6 days
+        $lock = Cache::lock('vendor_oui_db_refresh', 86400 * $this->min_refresh_days);
```

Comment:
Should use https://laravel.com/docs/10.x/scheduling#preventing-task-overlaps instead

Reply:
@Jellyfrog Already did, it didn't do what I wanted.
->withoutOverlapping() waits until other workers have finished, then runs.  I want it to ->runOnOneServer() and prevent (or at least disincentivize) users from running the fetch command too frequently.